ts/features/bonus/cgn/screens/CgnDetailScreen.tsx

```diff
@@ -262,6 +266,23 @@ const CgnDetailScreen = (props: Props): React.ReactElement => {
             </View>
           </Animated.ScrollView>
           <SectionStatusComponent sectionKey={"cgn"} />
+          {CardExpired.is(props.cgnDetails) && (
+            <GradientBottomActions
+              primaryActionProps={{
+                color: "danger",
+                label: I18n.t("bonus.cgn.activation.deactivate.expired"),
+                onPress: requestUnsubscription
+              }}
+              transitionAnimStyle={footerGradientOpacityTransition}
+              dimensions={{
+                bottomMargin,
+                extraBottomMargin: 0,
+                gradientAreaHeight,
+                spaceBetweenActions,
+                safeBackgroundHeight: bottomMargin
+              }}
+            />
+          )}
```

Comment:
This check must be done as the "else-case" of the condition which is present at line `286`, because in a View cannot coexist two or more GradientBottomActions or we could face in the bug that you can see in the following video:

https://github.com/user-attachments/assets/c437e0f8-46d8-4bf8-b904-7d78c4456fcb

What do you think to create a local component outside the render method, calling it for example: `FooterActions` where inside of it you check if the card is expired, if it's expired so render this bottom actions, otherwise do the checks which are present from line `286` to `318`

Reply:
I would not create a separate component right now since there is the task PE-610 where the bottom actions will have another structure

Reply:
I still need to extract a component to respect snarling rules